app/util/process_utils.py

```diff
@@ -23,3 +25,26 @@ def kill_gracefully(process, timeout=2):
         stdout, stderr = process.communicate()
 
     return process.returncode, stdout, stderr
+
+
+def Popen_with_delayed_expansion(cmd, *args, **kwargs):
```

Comment:
Do you think we would want to use this wrapper everywhere throughout the app? If so, we might want to just name it Popen and maybe enforce that this wrapper must be used instead of subprocess.Popen directly.